code/framework/encryption/src/main/java/io/cattle/platform/framework/encryption/Hasher.java

```diff
@@ -2,13 +2,21 @@
 
 import io.cattle.platform.util.type.Named;
 import io.github.ibuildthecloud.gdapi.model.Transformer;
+import io.cattle.platform.api.auth.LocalAuthPasswordValidator;
+import io.cattle.platform.json.JsonMapper;
+
+import javax.inject.Inject;
 
 import org.apache.commons.lang3.StringUtils;
 
 public abstract class Hasher implements Transformer, Named {
 
+    @Inject
+    JsonMapper jsonMapper;
+
     @Override
     public String transform(String value) {
+        LocalAuthPasswordValidator.validatePassword(value, jsonMapper);
```

Comment:
This seems like way too generic of a place to put this validation.  What use case are you trying to catch with this.